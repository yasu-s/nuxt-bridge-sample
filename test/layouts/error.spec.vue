<script lang="ts">
import { shallowMount } from '@vue/test-utils'
import ErrorLayout from '@/layouts/error.vue'

describe('@/layouts/error.vue', () => {
  test('is a Vue instance', () => {
    const wrapper = shallowMount(ErrorLayout, { propsData: { error: { statusCode: 404 } } })
    expect(wrapper.vm).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('statusCode=500', () => {
    const wrapper = shallowMount(ErrorLayout, { propsData: { error: { statusCode: 500 } } })
    expect(wrapper.find('h1').text()).toBe('An error occurred')
  })

  test('changeOtherErrorMessage', async () => {
    const wrapper = shallowMount(ErrorLayout, { propsData: { error: { statusCode: 500 } } })

    // eslint-disable-next-line
    const vm = wrapper.vm as any

    // eslint-disable-next-line
    vm.changeOtherErrorMessage('hoge')

    // eslint-disable-next-line
    await vm.$nextTick()

    expect(wrapper.find('h1').text()).toBe('hoge')
  })

  test('isAdmin=true', async () => {
    const wrapper = shallowMount(ErrorLayout, { propsData: { error: { statusCode: 500 } } })
    await wrapper.setData({ isAdmin: true })

    const actual = wrapper.find('.admin-label')
    expect(actual.exists()).toBeTruthy()
    expect(actual.text()).toBe('管理者')

    // eslint-disable-next-line
    expect((wrapper.vm as any).isAdmin).toBeTruthy()
  })
})
</script>
