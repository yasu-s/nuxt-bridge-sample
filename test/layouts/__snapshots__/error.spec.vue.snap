// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@/layouts/error.vue is a Vue instance 1`] = `
"<v-app-stub dark=\\"true\\" id=\\"app\\">
  <h1>
    404 Not Found
  </h1> <a> Home page </a>
  <!---->
</v-app-stub>"
`;
